packages/react-router/lib/router/define-route.ts

```diff
@@ -0,0 +1,69 @@
+import type { ReactNode } from "react";
+
+interface Context {} // TODO: AppLoadContext?
+
+type MaybePromise<T> = T | Promise<T>;
+
+type Serializable =
```

Comment:
How could we make this `Serializable` ?
```ts
type Foo = {
  k: readonly string[]
}
const foo: Foo = {k: ['v']}
const bar: Serializable = foo
```

```
Type 'Foo' is not assignable to type 'Serializable'.
  Type 'Foo' is not assignable to type '{ [key: string]: Serializable; [key: number]: Serializable; [key: symbol]: Serializable; }'.
    Property 'k' is incompatible with index signature.
      Type 'readonly string[]' is not assignable to type 'Serializable'.
        Type 'readonly string[]' is not assignable to type '{ [key: string]: Serializable; [key: number]: Serializable; [key: symbol]: Serializable; }'.
          Index signature for type 'string' is missing in type 'readonly string[]'.ts(2322)
```

Without `readonly`, it works fine.

Reply:
Yea that's a bug. We'll fix it.